catppuccin.tmux

```diff
@@ -1,23 +1,159 @@
 #!/usr/bin/env bash
 CURRENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-get-tmux-option() {
-  local option value default
-  option="$1"
-  default="$2"
-  value="$(tmux show-option -gqv "$option")"
-
-  if [ -n "$value" ]; then
-    echo "$value"
-  else
-    echo "$default"
-  fi
-}
-
 main() {
+  get-tmux-option() {
+    local option value default
+    option="$1"
+    default="$2"
+    value="$(tmux show-option -gqv "$option")"
+
+    if [ -n "$value" ]; then
+      echo "$value"
+    else
+      echo "$default"
+    fi
+  }
+
+  set() {
+    local option=$1
+    local value=$2
+    tmux set-option -gq "$option" "$value"
+  }
+
+  setw() {
+    local option=$1
+    local value=$2
+    tmux set-window-option -gq "$option" "$value"
+  }
+
   local theme
   theme="$(get-tmux-option "@catppuccin_flavour" "mocha")"
-  tmux run -b "$CURRENT_DIR/catppuccin-${theme}.tmuxtheme"
+
+  # NOTE: Pulling in the selected theme by the theme that's being set as local
+  # variables.
+  sed -E 's/^(.+=)/local \1/' \
+      > catppuccin-selected-theme.tmuxtheme \
+      < "${CURRENT_DIR}/catppuccin-${theme}.tmuxtheme"
+
+  source catppuccin-selected-theme.tmuxtheme
+
+  # status
+  set status "on"
+  set status-bg "${thm_bg}"
+  set status-justify "left"
+  set status-left-length "100"
+  set status-right-length "100"
+
+  # messages
+  set message-style "fg=${thm_cyan},bg=${thm_gray},align=centre"
+  set message-command-style "fg=${thm_cyan},bg=${thm_gray},align=centre"
+
+  # panes
+  set pane-border-style "fg=${thm_gray}"
+  set pane-active-border-style "fg=${thm_blue}"
+
+  # windows
+  setw window-status-activity-style "fg=${thm_fg},bg=${thm_bg},none"
+  setw window-status-separator ""
+  setw window-status-style "fg=${thm_fg},bg=${thm_bg},none"
+
+  # --------=== Statusline
+
+  # NOTE: Checking for the value of @catppuccin_window_tabs_enabled
+  wt_enabled="$(get-tmux-option "@catppuccin_window_tabs_enabled" "off")"
+  readonly wt_enabled
+
+  # NOTE: Checking for the value of @catppuccin_window_tabs_enabled
+  wi_enabled="$(get-tmux-option "@catppuccin_window_icons_enabled" "off")"
+  readonly wi_enabled
+
+  # README: Any variables marked `readonly` are the defaults so that it is
+  # easier to understand what is a default value. Some variables are marked
+  # `local` and that's because they will be used in place for the `setw` and
+  # `set` commands to make it easier to parse what is happening.
+
+  # Default Nerd Font zoom/mark icons
+  readonly icon_window_current="  "
+  readonly icon_window_last="  "
+  readonly icon_window_activity="  "
+  readonly icon_window_bell=" "
+  readonly icon_window_silent=" "
+  readonly icon_window_zoom="  "
+  readonly icon_window_mark=" "
```

Comment:
[=> I took the values of all of these from this SuperUser page here.](https://superuser.com/questions/1486785/how-to-customize-tmux-last-window-marker)

[=> Official Tmux man page, you'll have to search for the term **Status line**](https://man7.org/linux/man-pages/man1/tmux.1.html)

```
           Symbol    Meaning
           *         Denotes the current window.
           -         Marks the last window (previously selected).
           #         Window activity is monitored and activity has been
                                detected.
           !         Window bells are monitored and a bell has occurred
                                in the window.
           ~         The window has been silent for the monitor-silence
                                interval.
           M         The window contains the marked pane.
           Z         The window's active pane is zoomed.
```

Reply:
@sdemura is this something that you're expecting to see in this status line update?